### DISCUSSION TITLE
Disable auto-completion of quotes, braces etc. in JSON

### DISCUSSION THREAD

**Discussion by emz00** [Kong/insomnia#3670] [Ideas]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when a text editor gets in the way of me typing whatever I want. When I type `"` I want `"` to appear - nothing more, nothing less. Same for `{`. Auto-completion is extremely frustrating.

**Describe the solution you'd like**
An option to disable auto-completion. (If such an option already exist make it more obvious, because I cannot find it.)

**Describe alternatives you've considered**
- Not using Insomnia at all.
- Using Insomnia to send requests, but a real text editor to write them.

**Comment by BenoitHiller** [comment]
So my understanding is that auto-completing these identifiers is a bug caused by trying to not break syntax highlighting as the user types. There is no benefit to the completions, in the intended usage the user types over them anyway to advance the cursor, so it may as well not have bothered. Whereas there are a large number of typical edit operations where it does the wrong thing.

I understand that the more robust solutions for highlighting stability are a bit much to implement, but just a toggle would help me a lot even if it had significant downsides.

**Comment by emz00** [comment]
The [issue](https://github.com/Kong/insomnia/issues/3334) was closed as completed by @wdawson on Jul 1, 2021 without any comment, but this is still a problem as of Insomnia 2022.7.0. It's still really annoying, and I cannot see any way to disable it. I've tried setting "Autocomplete popup delay (ms)" to 0, I've tried changing the request body type from "JSON" to "Plain" - it still types `[]` when I type `[`. So was this option added somewhere that I cannot find?

**Comment by davehouser1** [comment]
Any update on this?
